Description: Disable buggy TokuDB tests
 Backported fix done by upstream that will be included
 in next release (10.0.15).
 Certain TokuDB tests fail because the Tokutek read free replication patch was
 not merged into MariaDB 10.0.14. These tests should be disabled until the patch
 is reviewed and accepted into MariaDB.
 This patch can be removed once upstream 10.0.15 is released and imported.
Author: Otto Kekäläinen <otto@seravo.fi>
Bug: https://mariadb.atlassian.net/browse/MDEV-6815
Last-Update: 2014-10-01

=== modified file 'storage/tokudb/CMakeLists.txt'
--- old/storage/tokudb/CMakeLists.txt	2014-09-23 20:03:35 +0000
+++ new/storage/tokudb/CMakeLists.txt	2014-09-25 17:00:41 +0000
@@ -18,7 +18,7 @@
 ENDIF()
 
 ############################################
-SET(TOKUDB_VERSION "7.1.8")
+SET(TOKUDB_VERSION "7.5.0")
 SET(TOKUDB_DEB_FILES "usr/lib/mysql/plugin/ha_tokudb.so\netc/mysql/conf.d/tokudb.cnf\nusr/bin/tokuftdump\nusr/share/doc/mariadb-server-10.0/README-TOKUDB\nusr/share/doc/mariadb-server-10.0/README.md" PARENT_SCOPE)
 SET(USE_BDB OFF CACHE BOOL "")
 MARK_AS_ADVANCED(BUILDNAME)

=== added file 'storage/tokudb/mysql-test/rpl/disabled.def'
--- old/storage/tokudb/mysql-test/rpl/disabled.def	1970-01-01 00:00:00 +0000
+++ new/storage/tokudb/mysql-test/rpl/disabled.def	2014-09-25 17:00:41 +0000
@@ -0,0 +1,12 @@
+rpl_tokudb_delete_pk: unreliable, uses timestamp differences
+rpl_tokudb_delete_pk_lookup1: unreliable, uses timestamp differences
+rpl_tokudb_update_pk_uc0_lookup0: unreliable, uses timestamp differences
+rpl_tokudb_update_pk_uc0_lookup1: unreliable, uses timestamp differences
+rpl_tokudb_update_pk_uc1_lookup0: unreliable, uses timestamp differences
+rpl_tokudb_update_pk_uc1_lookup1: unreliable, uses timestamp differences
+rpl_tokudb_update_unique_uc0_lookup0: unreliable, uses timestamp differences
+rpl_tokudb_update_unique_uc0_lookup1: unreliable, uses timestamp differences
+rpl_tokudb_write_pk: unreliable, uses timestamp differences
+rpl_tokudb_write_pk_uc1: unreliable, uses timestamp differences
+rpl_tokudb_write_unique: unreliable, uses timestamp differences
+rpl_tokudb_write_unique_uc1: unreliable, uses timestamp differences

=== modified file 'storage/tokudb/mysql-test/tokudb/disabled.def'
--- old/storage/tokudb/mysql-test/tokudb/disabled.def	2014-09-23 20:03:35 +0000
+++ new/storage/tokudb/mysql-test/tokudb/disabled.def	2014-09-25 17:00:41 +0000
@@ -26,3 +26,5 @@
 mvcc-19: how this could work, if alter needs an exclusive mdl lock?
 mvcc-20: how this could work, if alter needs an exclusive mdl lock?
 cluster_key_part: engine options on partitioned tables
+i_s_tokudb_lock_waits_released: unstable, race conditions
+i_s_tokudb_locks_released: unstable, race conditions

=== modified file 'storage/tokudb/mysql-test/tokudb/t/i_s_tokudb_lock_waits_released.test'
--- old/storage/tokudb/mysql-test/tokudb/t/i_s_tokudb_lock_waits_released.test	2014-09-12 22:28:15 +0000
+++ new/storage/tokudb/mysql-test/tokudb/t/i_s_tokudb_lock_waits_released.test	2014-09-25 17:00:41 +0000
@@ -1,8 +1,4 @@
 # verify that information_schema.tokudb_locks gets populated with locks, information_schema.tokudb_lock_waits gets
-if (`select @@tokudb_version <= "7.1.8"`)
-{
-  --skip Race condition in the test in TokuDB 7.1.8 or earlier
-}
 # populated with 1 lock_wait and all transactions are present in information_schema.tokudb_trx for 2 clients
 
 source include/have_tokudb.inc;

=== modified file 'storage/tokudb/mysql-test/tokudb/t/i_s_tokudb_locks_released.test'
--- old/storage/tokudb/mysql-test/tokudb/t/i_s_tokudb_locks_released.test	2014-09-12 22:28:15 +0000
+++ new/storage/tokudb/mysql-test/tokudb/t/i_s_tokudb_locks_released.test	2014-09-25 17:00:41 +0000
@@ -1,7 +1,3 @@
-if (`select @@tokudb_version <= "7.1.8"`)
-{
-  --skip Race condition in the test in TokuDB 7.1.8 or earlier
-}
 # verify that information_schema.tokudb_locks gets populated with locks for 2 clients
 
 source include/have_tokudb.inc;

